distribution/sc-proxy-dex/src/proxy_farm.rs

```diff
@@ -252,60 +243,112 @@ pub trait ProxyFarmModule:
         let new_wrapped_farm_token_attributes = WrappedFarmTokenAttributes {
             farm_token_id: new_farm_token_id,
             farm_token_nonce: new_farm_token_nonce,
+            farm_token_amount: new_farm_token_total_amount.clone(),
             farming_token_id: wrapped_farm_token_attrs.farming_token_id,
             farming_token_nonce: wrapped_farm_token_attrs.farming_token_nonce,
+            farming_token_amount: self.rule_of_three_non_zero_result(
+                &amount,
+                &wrapped_farm_token_attrs.farm_token_amount,
+                &wrapped_farm_token_attrs.farming_token_amount,
+            )?,
         };
-        self.create_and_send_wrapped_farm_tokens(
+        self.create_by_wrapped_farm_tokens_by_merging_and_send(
             &new_wrapped_farm_token_attributes,
             &new_farm_token_total_amount,
+            &farm_address,
             &caller,
-        );
+        )?;
         self.nft_burn_tokens(&token_id, token_nonce, &amount);
 
         Ok(())
     }
 
-    fn get_attributes(
+    #[payable("*")]
+    #[endpoint(compoundRewardsProxy)]
+    fn compound_rewards_proxy(
         &self,
-        token_id: &TokenIdentifier,
-        token_nonce: Nonce,
-    ) -> SCResult<WrappedFarmTokenAttributes> {
-        let token_info = self.blockchain().get_esdt_token_data(
-            &self.blockchain().get_sc_address(),
-            token_id,
-            token_nonce,
+        #[payment_token] payment_token_id: TokenIdentifier,
+        #[payment_nonce] payment_token_nonce: Nonce,
+        #[payment_amount] payment_amount: Self::BigUint,
+        farm_address: Address,
+    ) -> SCResult<()> {
+        self.require_is_intermediated_farm(&farm_address)?;
+        self.require_wrapped_farm_token_id_not_empty()?;
+        self.require_wrapped_lp_token_id_not_empty()?;
+        self.require_deposit_empty_or_tokens_are_wrapped_farm_tokens()?;
+        require!(payment_amount != 0, "Payment amount cannot be zero");
+        let wrapped_farm_token = self.wrapped_farm_token_id().get();
+        require!(
+            payment_token_id == wrapped_farm_token,
+            "Should only be used with wrapped farm tokens"
         );
 
-        let attributes = token_info.decode_attributes::<WrappedFarmTokenAttributes>();
-        match attributes {
-            Result::Ok(decoded_obj) => Ok(decoded_obj),
-            Result::Err(_) => {
-                return sc_error!("Decoding error");
-            }
-        }
-    }
+        let wrapped_farm_token_attrs =
+            self.get_wrapped_farm_token_attributes(&payment_token_id, payment_token_nonce)?;
+        let farm_token_id = wrapped_farm_token_attrs.farm_token_id;
+        let farm_token_nonce = wrapped_farm_token_attrs.farm_token_nonce;
+        let farm_amount = payment_amount.clone();
 
```

Comment:
The compound function only works for the lockedMEX. Is it better to check if the farming_token_id = locked_asset_id here?